test/e2e/networkpolicy_test.go

```diff
@@ -217,15 +239,16 @@ func (data *TestData) testDifferentNamedPorts(t *testing.T) {
 			},
 		}},
 	}
-	np, err := data.createNetworkPolicy("test-networkpolicy-allow-client0-to-http", spec)
+	np, err := data.createNetworkPolicy(randName("test-networkpolicy-allow-client0-to-http"), spec)
 	if err != nil {
 		t.Fatalf("Error when creating network policy: %v", err)
 	}
-	defer func() {
+	cleanupFuncs = append(cleanupFuncs, func() {
 		if err = data.deleteNetworkpolicy(np); err != nil {
 			t.Fatalf("Error when deleting network policy: %v", err)
 		}
-	}()
+	})
+	time.Sleep(networkPolicyDelay)
```

Comment:
Not sure if it's changing the podSelector to "MatchExpressions" caused the networkpolicy takes a little more time to realize. The test now has a chance to fail without the sleep, I can confirm it's not because of the conversion function change as it also failed when I just changed the test code in this PR: https://github.com/vmware-tanzu/antrea/runs/1431829031?check_suite_focus=true